tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/ops/TransformToGrayscaleOp.java

```diff
@@ -0,0 +1,141 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+package org.tensorflow.lite.support.image.ops;
+
+import android.graphics.Canvas;
+import android.graphics.Bitmap;
+import android.graphics.Paint;
+import android.graphics.ColorMatrixColorFilter;
+import android.graphics.ColorFilter;
+import android.graphics.PointF;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.tensorflow.lite.support.image.ImageOperator;
+import org.tensorflow.lite.support.image.TensorImage;
+import org.tensorflow.lite.support.tensorbuffer.TensorBuffer;
+import org.tensorflow.lite.DataType;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import static org.tensorflow.lite.support.image.ColorSpaceType.GRAYSCALE;
+
+/**
+ * As a computation unit for processing images, it could transform a colored image to grayscale.
+ *
+ * The conversion is based on OpenCV RGB to GRAY conversion
+ * https://docs.opencv.org/master/de/d25/imgproc_color_conversions.html#color_convert_rgb_gray
+ *
+ * The 3 channels of the grayscale image are identical and one of them is used to create the TensorBuffer
+ *
+ */
+public class TransformToGrayscaleOp implements ImageOperator {
+
+    /**
+     * Creates a TransformToGrayscaleOp
+     *
+     * No parameters are used
+     */
+    public TransformToGrayscaleOp() {
+    }
+
+    /**
+     * Applies the defined transformation to grayscale and returns a TensorImage
+     *
+     * Because the 3 channels of the image are identical only one is used to create the TensorBuffer
+     *
+     * @param image input image.
+     * @return output image.
+     */
+    @Override
+    @NonNull
+    public TensorImage apply(@NonNull TensorImage image) {
```

Comment:
We also need to check the color type of the image here.

1. RGB -> good
2. Grayscale -> return and skip
3. Others -> exception (and document in javadoc)

Reply:
Done